V2WebSocketProtocolGuide.md

```diff
@@ -138,23 +138,51 @@ Local proxy will initiate a web socket handshake to connect to the tunnel, using
+7. This will mark _streamID_ 1 for _serviceID_ SSH2 as inactive and the source local proxy will send another stream_start message with _streamID_ 2 as part of the reset process.
+8. After this the new mapping will be:  _serviceID_ SSH1 -> active _streamID_ 1 and  _serviceID_ SSH2 -> active _streamID_ 2.
+9. Every data message received for SSH2 that has a _streamID_ of 1 thereafter will be ignored.
+
 ####  Step 3: End to end data transfer over the tunnel 
 
 On receiving a StreamStart, the destination local proxy will update the service ID -> Stream ID mapping and connect to the configured destination service for a service ID.  The destination local proxy does not send a reply to the source local proxy on successful connection. Immediately after the source local proxy sends StreamStart and immediately after the destination establishes a valid TCP connection, each side respectively can begin to send and receive incoming messages on the active data stream. When the data stream is closed or disrupted (for the local proxy, this is a TCP close or I/O error on the TCP socket), a StreamReset control message with the currently stored stream ID and its service ID should be sent through the tunnel so the tunnel peer can react appropriately and end the data stream. Control messages associated with a stream should be processed with the same stream ID filter.
 
 Here are some important things to know for a high-level understanding of tunneling data stream handling:
 
--   The service may use the Service ID to decide how to route traffic between connected tunnel clients. 
-      -   For example,  when local proxy received a data packet with Service ID  SSH1, it will look up the configuration for SSH1 and see which port this service ID is mapped to. If SSH1 is mapped to port 22 on local host, then this data packet will be forward to port 22 on local host. 
--   The local proxy uses the service ID -> stream ID mapping to check the current active stream ID for a specific service ID. 
--   The stream ID validation for a certain stream(service ID) will only be performed on message type _StreamReset_ and _Data_. If a received message failed the stream ID validation, this message is considered to be stale and will be discarded by local proxy. 
--   The local proxy, and library clients may use stream ID to determine how to respond to or filter incoming messages
+- The service may use the Service ID to decide how to route traffic between connected tunnel clients. 
+      - For example,  when local proxy received a data packet with Service ID  SSH1, it will look up the configuration for SSH1 and see which port this service ID is mapped to. If SSH1 is mapped to port 22 on local host, then this data packet will be forward to port 22 on local host. 
+- A stream start message may include one of the service ID's defined as part of the tunnel, or no service ID. It may not, however, include any other service ID not defined during tunnel creation.
```

Comment:
If a stream is started without a service id and then a subsequent message is sent with a service id, the secure tunnel service closes the socket. As such, I think the destination device should close all streams using service ids if it receives a stream start without a service id (and treat the current source device as V1) and in the event there is a V1 connection and a stream start comes in with a service id, it should close the V1 stream and assume we are now using V2.